core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1276,6 +1277,32 @@ public boolean isNeededDependenciesCompatibleWithInstalledVersion(PluginManager.
             return getDependenciesIncompatibleWithInstalledVersion(cache).isEmpty();
         }
 
+        /**
+         * Returns true if and only if this update addressed a currently active security vulnerability.
+         *
+         * @return true if and only if this update addressed a currently active security vulnerability.
+         */
+        @Restricted(NoExternalUse.class) // Jelly
+        public boolean fixesSecurityVulnerabilities() {
+            final PluginWrapper installed = getInstalled();
+            if (installed == null) {
+                return false;
+            }
+            boolean allWarningsStillApply = true;
+            for (Warning warning : ExtensionList.lookupSingleton(UpdateSiteWarningsMonitor.class).getActivePluginWarningsByPlugin().getOrDefault(installed, Collections.emptyList())) {
+                boolean thisWarningApplies = false;
+                for (WarningVersionRange range : warning.versionRanges) {
+                    if (range.includes(new VersionNumber(version))) {
```

Comment:
NIT: Maybe this `VersionNumber` object should be constructed outside the loop. Although java might just optimize this away